src/overlays/actors/ovl_Bg_Haka_Trap/z_bg_haka_trap.c

```diff
@@ -40,36 +41,490 @@ const ActorInit Bg_Haka_Trap_InitVars = {
     (ActorFunc)BgHakaTrap_Update,
     (ActorFunc)BgHakaTrap_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Trap/BgHakaTrap_Init.s")
+static ColliderCylinderInit sCylinderInit = {
+    { COLTYPE_METAL_SHIELD, 0x11, 0x0D, 0x09, 0x20, COLSHAPE_CYLINDER },
+    { 0x00, { 0xFFCFFFFF, 0x00, 0x04 }, { 0xFFCFFFFF, 0x00, 0x00 }, 0x01, 0x01, 0x01 },
+    { 30, 90, 0, { 0, 0, 0 } },
+};
+
+static ColliderTrisItemInit sTrisItemsInit[2] = {
+    {
+        { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00020000, 0x00, 0x00 }, 0x00, 0x01, 0x00 },
+        { { { 1800.0f, 1200.0f, 0.0f }, { -1800.0f, 1200.0f, 0.0f }, { -1800.0f, 0.0f, 0.0f } } },
+    },
+    {
+        { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00020000, 0x00, 0x00 }, 0x00, 0x01, 0x00 },
+        { { { 1800.0f, 1200.0f, 0.0f }, { -1800.0f, 0.0f, 0.0f }, { 1800.0f, 0.0f, 0.0f } } },
+    },
+};
+
+static ColliderTrisInit sTrisInit = {
+    { COLTYPE_UNK10, 0x00, 0x09, 0x00, 0x20, COLSHAPE_TRIS },
+    2,
+    sTrisItemsInit,
+};
+
+static CollisionCheckInfoInit sColChkInfoInit = { 0x00, 0x0050, 0x0064, 0xFF };
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+static UNK_TYPE D_80881014 = 0;
+static UNK_TYPE D_80881018 = 0;
+
+static Vec3f D_8088101C = { 0.0f, 0.0f, 0.0f };
+
+static Gfx* sDLists[5] = {
+    0x06007610, 0x06009860, 0x06007EF0, 0x06008A20, 0x060072C0,
+};
+
+static Color_RGBA8 D_8088103C = { 0 };
+
+typedef enum {
+    /* 0x00 */ TRAP_GUILLOTINE_SLOW,
+    /* 0x01 */ TRAP_SPIKED_BOX,
+    /* 0x02 */ TRAP_SPIKED_WALL,
+    /* 0x03 */ TRAP_SPIKED_WALL_2,
+    /* 0x04 */ TRAP_PROPELLER,
+    /* 0x05 */ TRAP_GUILLOTINE_FAST,
+} TrapType;
+
+extern UNK_TYPE D_060081D0;
+extern UNK_TYPE D_06008D10;
+extern UNK_TYPE D_06009CD0;
+
+void BgHakaTrap_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgHakaTrap* this = THIS;
+    s32* temp_ptr;
+    s32 sp2C = 0;
+
+    Actor_ProcessInitChain(thisx, sInitChain);
+    thisx->params &= 0xFF;
+
+    if (thisx->params != TRAP_PROPELLER) {
+        Collider_InitCylinder(globalCtx, &this->colliderCylinder);
+        Collider_SetCylinder(globalCtx, &this->colliderCylinder, thisx, &sCylinderInit);
+
+        temp_ptr = &D_80881014;
+        if ((thisx->params == TRAP_GUILLOTINE_SLOW) || (thisx->params == TRAP_GUILLOTINE_FAST)) {
+            this->actionTimer = 20;
+            this->colliderCylinder.dim.yShift = 10;
+            thisx->velocity.y = 0.1f;
+
+            if (thisx->params == TRAP_GUILLOTINE_FAST) {
+                thisx->params = TRAP_GUILLOTINE_SLOW;
+                this->unk_16A = 1;
+            }
+
+            this->actionFunc = &func_80880484;
```

Comment:
We don't use `&` for function pointers.
```suggestion
            this->actionFunc = func_80880484;
```

Reply:
fixed